src/pages/import/components/ImportTimetable.tsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from 'react';
+
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+  useToast,
+  VStack,
+  Text,
+  Alert,
+  AlertIcon,
+  useMediaQuery,
+} from '@chakra-ui/react';
+import { BsPlusCircleFill } from 'react-icons/bs';
+import { useNavigate } from 'react-router';
+
+import { Timetable } from 'lib/fetchers';
+import { SavedCourse, useSavedCourses } from 'lib/hooks/useSavedCourses';
+
+import { SelectedCoursesCardList } from 'pages/scheduler/components/share/SelectedCoursesCardList';
+
+export function ImportTimetable({ loading, data }: { loading: boolean; data: Timetable }): JSX.Element {
+  const { importCourses } = useSavedCourses();
+  const toast = useToast();
+  const [isSmallScreen] = useMediaQuery('(min-width:680px)');
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const filteredCoursesList: SavedCourse[] = [];
+  const [filteredCourses, setFilteredCourses] = useState(filteredCoursesList);
+  const [timetableTerm, setTimetableTerm] = useState('');
+  const navigate = useNavigate();
+
+  const handleImport = () => {
+    importCourses(data.courses, data.term);
+    onClose();
+    navigate(`/scheduler/${timetableTerm}`);
+    toast({
+      title: 'Timetable imported!',
+      status: 'success',
+      duration: 3000,
+    });
+  };
+
+  useEffect(() => {
+    if (data && !loading) {
+      data.courses.forEach(({ subject, code, pid, lecture, lab, tutorial, color }) => {
+        const newCourse: SavedCourse = {
+          subject,
+          pid,
+          code,
+          term: data.term,
+          selected: true,
+          showSections: true,
+          lecture: lecture ? lecture[0] : undefined,
+          lab: lab ? lab[0] : undefined,
+          tutorial: tutorial ? tutorial[0] : undefined,
+          color: color,
```

Comment:
```suggestion
          color,
```
nit: automatically assigns variables of same name